nltk/model/counter.py

```diff
@@ -0,0 +1,99 @@
+# Natural Language Toolkit: Language Model Counters
+#
+# Copyright (C) 2001-2016 NLTK Project
+# Author: Ilia Kurenkov <ilia.kurenkov@gmail.com>
+# URL: <http://nltk.org/>
+# For license information, see LICENSE.TXT
+
+from __future__ import unicode_literals
+
+from collections import Counter, defaultdict
+from itertools import chain
+
+from nltk.util import ngrams
+from nltk.probability import FreqDist, ConditionalFreqDist
+from nltk import compat
+
+
+@compat.python_2_unicode_compatible
+class LanguageModelVocabulary(Counter):
+    """Stores language model vocabulary.
+
+    Satisfies two common language modeling requirements for a vocabulary:
+    - When checking membership and calculating its size, filters items by comparing
+      their counts to a cutoff value.
+    - Adds 1 to its size so as to account for "unknown" tokens.
+    """
+
+    def __init__(self, unknown_cutoff, *counter_args):
+        Counter.__init__(self, *counter_args)
```

Comment:
I would say, `super()` here makes sense otherwise, `class LanguageModelVocabulary(Counter)` would be redundant since we can do `class LanguageModelVocabulary(object)` and still do `Counter.__init__(self, *counter_args)`. But I think someone else from a more pythonic background should comment on this.

As for backwards compat, I guess so, NLTK still supports Python2.7, so we have to use old style `super()`, e.g. https://github.com/nltk/nltk/blob/develop/nltk/compat.py#L684